### PR TITLE
do not export as.Date.IDate anymore

### LINKED ISSUE #4777: S3 method lookup found 'as.Date.IDate' on search path
One of [CRAN machines reports](https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-debian-clang/data.table-00check.html) an issue caused by a workaround implemented to resolve https://github.com/Rdatatable/data.table/issues/1500 in 4f6908af4efee3a23bee27a8b8e51ae716d3941b

```
  S3 method lookup found 'as.Date.IDate' on search path
  S3 method lookup found 'as.Date.IDate' on search path
```
According to changelog of zoo, their as.Date method has been changed in 1.8-1 (Jan 2018), so it is likely that workaround is no longer needed.

### PR DESCRIPTION
**Pull request by jangorecki**
closes #4777

### CODE REVIEW

**Comment by MichaelChirico** [comment]
FWIW i've got a mirror of CRAN locally, statically checked if any revdeps (recursive depends/imports) were using the previous export, didn't find any.

**Comment by eddelbuettel** [comment]
FWIW comes up too when letting GitHub search in the CRAN mirror:  https://github.com/search?q=org%3Acran+as.Date.IDate

File: NAMESPACE [hunk 1]
```diff
@@ -151,8 +151,11 @@ S3method("+", IDate)
 S3method("-", IDate)
 S3method(as.character, ITime)
 S3method(as.data.frame, ITime)
-S3method(as.Date, IDate)   # note that zoo::as.Date masks base::as.Date. Both generic.
-export(as.Date.IDate)      # workaround for zoo bug, see #1500. Removing this export causes CI pipeline to fail on others.Rraw test 6, but I can't reproduce locally.
+if (getRversion() >= "3.6.0") {
+  S3method(as.Date, IDate)
```
Comment (mattdowle) [inline NAMESPACE:155]: @jangorecki Why >= "3.6.0"?  Won't `as.Date(x)` break in R < 3.6.0 if the method is not exported?
Also could the comments be put back so we know there's some history on these lines pointing to #1500.

**Review by mattdowle — CHANGES_REQUESTED** [review]

File: NAMESPACE (hunk 1 above)
Comment (mattdowle) [reply NAMESPACE:155]: Higher up in the file the switch on 3.6.0 is about delayed registration (i.e. the `pkg::` prefix on the first argument to `S3method`) but that's not the case here with `as.Date`.

Comment (mattdowle) [reply NAMESPACE]: I figured the >=3.6 was copied from higher up. So I just went ahead and removed it, to merge and let's see if it passes CI. If this was wrong then we can always do a follow up PR.

Comment (mattdowle) [reply NAMESPACE]: When I added the suggest version for zoo, I did bit64 and bit at the same time (although nothing to do with this PR).

**Review by mattdowle — APPROVED** [review]

**Pull request merged**